libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -264,12 +264,12 @@ bool AC_Fence::pre_arm_check_circle(const char* &fail_msg) const
 // additional checks for the alt fence:
 bool AC_Fence::pre_arm_check_alt(const char* &fail_msg) const
 {
-    if (_alt_max < 0.0f) {
+    if (!is_positive(_alt_max)) {
         fail_msg = "Invalid FENCE_ALT_MAX value";
         return false;
     }
 
-    if (_alt_min < -100.0f) {
+    if (is_zero(_alt_min)) {
```

Comment:
this is relative to home, so we can have negative value

Reply:
In theory then, you could have an `_alt_max < 0` if you want to fly with a ceiling below home.

Reply:
It might be best in both cases to do a check against active `FENCE_TYPE`, and if the ceiling/floor is set, then check if is zero?

eg. `get_enabled_fences() & AC_FENCE_TYPE_ALT_MIN`... however thinking about it... a fence floor/ceiling of zero could equally be valid.

Reply:
@khancyr san. That's right, the lowest altitude has a negative value. The lowest altitude of 0 hits the fence. Therefore, if it was set to 0, we assumed it was abnormal.

@joshanne san. The highest altitude was processed before, and anything less than zero is considered abnormal. The minimum altitude can have positive or negative values.The highest altitude was processed before, and anything less than zero is considered abnormal.
The minimum altitude can have positive or negative values.

Reply:
This check should probably just be removed.

The fixed -100 here is odd; sub users might want a fence 200m down, breaches blowing ballast tanks or whatnot.

I think the only thing we can really enforce with these two parameters is that _min < _max - Copters generally want to fly with _min and _max positive, Subs negative.